enclave/src/polkadex_orderbook_storage.rs

```diff
@@ -0,0 +1,76 @@
+use sgx_types::{sgx_epid_group_id_t, sgx_status_t, sgx_target_info_t, SgxResult};
+use std::collections::HashMap;
+use std::string::String;
+use std::sync::{
+    atomic::{AtomicPtr, Ordering},
+    Arc, SgxMutex, SgxMutexGuard,
+};
+
+use polkadex_primitives::types::Order;
+
+static GLOBAL_ORDERBOOK_STORAGE: AtomicPtr<()> = AtomicPtr::new(0 as *mut ());
+pub struct OrderbookStorage {
+    storage: HashMap<String, Order>,
+}
+
+impl OrderbookStorage {
+    pub fn create() -> OrderbookStorage {
+        OrderbookStorage {
+            storage: HashMap::new(),
+        }
+    }
+
+    /// Inserts a order_uid-order pair into the orderbook.
+    /// If the orderbook did not have this order_uid present, [None] is returned.
+    /// If the orderbook did have this order_uid present, the order is updated, and the old order is returned.
+    pub fn add_order(&mut self, order_uid: String, order: Order) -> Option<Order> {
+        self.storage.insert(order_uid, order)
+    }
+
+    /// Inserts a order_uid-order pair into the orderbook.
+    /// If the orderbook did not have this order_uid present, [None] is returned.
+    /// If the orderbook did have this order_uid present, the order is updated, and the old order is returned.
+    pub fn set_order(&mut self, order_uid: String, order: Order) -> Option<Order> {
```

Comment:
@Gauthamastro what is difference between this and the add_order function?

Reply:
Nothing since the underlying insert functions add if the key is not found and updates value if found. I just felt there might be a situation where the naming of the function can trouble us, hence the function.